### PR TITLE
Python: Add Tornado source modeling

### PR DESCRIPTION
**Pull request by RasmusWL**
A draft PR since it builds upon the changes from https://github.com/github/codeql/pull/4864.

Locally confirmed to bring back the missing result for https://lgtm.com/projects/g/ArchiveTeam/wpull?mode=tree.

### CODE REVIEW

File: python/ql/src/semmle/python/frameworks/Tornado.qll [hunk 1]
```diff
@@ -366,4 +436,84 @@ private module Tornado {
...
+   * A regex that is used to set up a route.
+   *
+   * Needs this subclass to be considered a RegexString.
+   */
+  private class TornadoRouteRegex extends RegexString {
+    TornadoRouteSetup setup;
+
+    TornadoRouteRegex() {
+      this instanceof StrConst and
+      DataFlow::localFlow(DataFlow::exprNode(this), setup.getUrlPatternArg())
```
Comment (yoff) [inline python/ql/src/semmle/python/frameworks/Tornado.qll:479]: Might be able to ask for a LocalSourceNode here.

File: python/ql/src/semmle/python/frameworks/Tornado.qll [hunk 2]
```diff
@@ -0,0 +1,543 @@
...
+    }
```
Comment (yoff) [inline python/ql/src/semmle/python/frameworks/Tornado.qll:480]: We might be able to ask for a LocalSourceNode here.

**Comment by yoff** [comment]
Preliminary review, since it is only a draft. But it basically looks great 👍 (nice, clean history too)

File: python/ql/src/semmle/python/frameworks/Tornado.qll (hunk 2 above)
Comment (RasmusWL) [reply python/ql/src/semmle/python/frameworks/Tornado.qll]: Yep, just rewrote it to match DjangoRouteRegex

**Comment by RasmusWL** [comment]
Basically nothing changed since your last "preliminary" review, now it's just rebased on top of main (meaning the UI is only listing the commits that are part of this PR).

Only new thing is using LocalSourceNode as you suggested :wink:

**Review by yoff — APPROVED** [review]
Looks good to merge now. Should the remaining failing tests be tracked in an issue or is the (awesome) fail column sufficient?

**Comment by RasmusWL** [comment]
> Looks good to merge now. Should the remaining failing tests be tracked in an issue or is the (awesome) fail column sufficient?

Good point. We probably should be tracking all those missing models in some issue :+1:

**Pull request merged**